### PR TITLE
Support date-range filtering in referrals fetch

### PR DESCRIPTION
**Pull request by kelvin-chappell**
This is the API change.  The UI change will follow.

### CODE REVIEW

File: app/controllers/CampaignApi.scala [hunk 1]
```diff
@@ -105,8 +109,15 @@ class CampaignApi(components: ControllerComponents, authAction: AuthAction[AnyCo
...
   }
 
-  def getCampaignReferrals(campaignId: String) = authAction { _ =>
-    CampaignReferralRepository.getCampaignReferrals(campaignId) match {
+  def getCampaignReferrals(campaignId: String, start: Option[String], end: Option[String]) = authAction { _ =>
+    def toDate(o: Option[String]): Option[LocalDate] = o flatMap { s =>
+      Try(LocalDate.parse(s)).toOption
+    }
+    val dateRange = for {
+      from <- toDate(start)
```
Comment (LATaylor-guardian) [inline app/controllers/CampaignApi.scala:117]: What happens if a date range is not specified? Will it show all in this case as expected?

**Comment by LATaylor-guardian** [comment]
👍

File: app/controllers/CampaignApi.scala (hunk 1 above)
Comment (kelvin-chappell) [reply app/controllers/CampaignApi.scala:117]: The idea is `dateRange` will then be `None`, which will mean the filter won't apply in `CampaignReferralRepository`, so then all the campaign figures will be fetched.

Comment (kelvin-chappell) [reply app/controllers/CampaignApi.scala:117]: So you have to give both dates if you want filtered results, which seems like a reasonable first iteration.

**Pull request merged**